### PR TITLE
#157: Fix Missing Page Titles

### LINKED ISSUE #157: Missing Page Titles
Loving the new direction!

Noticed there are missing page titles for each new page deployed, title is "Red Badger":

| URL | Page | Current Title |
|-----|---|-----|
https://red-badger.com/ | Home Page | Red Badger |
https://red-badger.com/what-we-do/ | What we do | Red Badger |

Leads to `Back Button` nav not being clear where you are.  Additionally may effect the listing of pages in Google.

### PR DESCRIPTION
**Pull request by jonsharratt**
### Motivation

- Closes #157 for missing page titles.  Had a spare 20 mins to take a quick look to fix the issue I raised.

### Test plan

* Go to home page, check page title `Home | Red Badger`
* Go to what we do, check page title `What we do | Red Badger`
* Go 404 page, check page title `Not found | Red Badger`

### Pre-merge checklist

- [ ] Unit tests

@lpil (Looking for guidance on this one, compiler `layoutTemplate` only returns body content, not sure I understand it fully... yet).

- [ ] Reviews
  - [ ] Code review
  - [ ] Design review
- [ ] Manual testing
  - [ ] Windows 7 IE 11
  - [ ] Windows 10 Edge
  - [x] Mac OS El Capitan Safari
  - [x] Mac OS El Capitan Chrome
  - [x] Mac OS El Capitan Firefox
  - [ ] iOS 9 Safari
  - [ ] Android 6 Chrome
- [ ] Tester approved

### CODE REVIEW

File: site/index.ejs [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <link rel="icon" href="/assets-honestly/favicons/favicon-96x96.png" type="image/png" sizes="96x96">
     <link rel="icon" href="/assets-honestly/favicons/favicon-192x192.png" type="image/png" sizes="192x192">
     <link rel="icon" href="/assets-honestly/favicons/favicon-194x194.png" type="image/png" sizes="194x194">
-    <title>Red Badger</title>
+    <title><% if (typeof title !== 'undefined') {%><%= title %><% } %></title>
```
Comment (zoeabryant) [inline site/index.ejs]: Should we have fallback here? @asavin

File: site/routes/index.js [hunk 2]
```diff
@@ -32,13 +32,19 @@ function prefixRoutes(rs) {
 }
 
 export default function routes({ contactUsURL }) {
+  const titleSuffix = process.env.COMPANY || 'Red Badger';
```
Comment (zoeabryant) [inline site/routes/index.js]: Not sure `COMPANY` really needs to be an environment variable, particularly as we have a fall back here. @asavin ?

**Comment by zoeabryant** [comment]
Thanks for the PR 👍

File: .env.example [hunk 3]
```diff
@@ -8,3 +8,4 @@ MAILCHIMP_API_KEY="YOUR-MAILCHIMP-API-KEY-HERE"
 SECRET_ENCRYPTION_KEY="YOUR-SECRET-KEY"
 MAILING_LIST_SERVICE_URL="The mailing list URL"
 MAILING_LIST_ID="The mailing list id"
+COMPANY="The company name"
```
Comment (lpil) [inline .env.example]: No reason to make this configurable imo.

**Comment by lpil** [comment]
This looks like it'll work for the static HTML, but the title will fall out of sync as the user navigates around the site with the front-end app. Good enough for SEO, but would be nice to get both.

**Comment by jonsharratt** [comment]
The front end app works also, I tested both `make dev-static` and front end with `make dev` is there another version to test @lpil ?

**Comment by asavin** [comment]
Needs testing cc @AndreiDanielIonescu @AndrewBestbier with and without JS.

**Comment by lpil** [comment]
@jonsharratt I thought it would be broken with `make dev`. How does the title get set dynamically when everything is done client side? I can't spot it.

**Comment by jonsharratt** [comment]
@lpil As far as I understood it, https://github.com/grahammendick/navigation does it for you pass title to it's stateContext 💃

**Comment by lpil** [comment]
Ah neat! Didn't know that :)

**Comment by asavin** [comment]
This looks good to me. Let's raise a separate PR purely for branch testing purposes and merge this one once the testing has passed.

**Pull request merged**